WooCommerce/Classes/ViewRelated/Products/Edit Product/Inventory Settings/ProductInventorySettingsViewController.swift

```diff
@@ -284,14 +284,24 @@ private extension ProductInventorySettingsViewController {
         }
         cell.configure(viewModel: cellViewModel)
 
-        // Configures accessory view for adding SKU from barcode scanner.
-        if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.barcodeScanner) {
+        // Configures accessory view for adding SKU from barcode scanner by fetching switch's state from app settings.
+        guard ServiceLocator.featureFlagService.isFeatureFlagEnabled(.productSKUInputScanner) && UIImagePickerController.isSourceTypeAvailable(.camera) else {
+            return
+        }
+        let action = AppSettingsAction.loadProductSKUInputScannerFeatureSwitchState() { [weak cell, self] result in
+            guard let cell = cell else { return }
+            guard let isEnabled = try? result.get(), isEnabled else {
+                return cell.accessoryView = nil
+            }
+
             let button = UIButton(type: .detailDisclosure)
             button.applyIconButtonStyle(icon: .scanImage)
-            button.addTarget(self, action: #selector(scanSKUButtonTapped), for: .touchUpInside)
-
+            button.on(.touchUpInside) { [weak self] _ in
+                self?.scanSKUButtonTapped()
+            }
```

Comment:
We can also use the native method `addAction`:
```suggestion
            button.addAction(UIAction { [weak self] _ in
                self?.scanSKUButtonTapped()
            }, for: .touchUpInside)
```

Also, I think we shoud also remove the `@objc` attribute for the method `scanSKUButtonTapped` 😄

Reply:
Great suggestion! TIL [iOS 14+ `UIControl.addAction` ](https://developer.apple.com/documentation/uikit/uicontrol/3600490-addaction) 😮 updated in https://github.com/woocommerce/woocommerce-ios/pull/5695/commits/452c9cc4e2496685d9950d659170bec9efa77478